src/mlpack/core/optimizers/parallel_sgd/parallel_sgd_impl.hpp

```diff
@@ -45,6 +45,10 @@ double ParallelSGD<DecayPolicyType>::Optimize(
   arma::Col<size_t> visitationOrder = arma::linspace<arma::Col<size_t>>(0,
       (function.NumFunctions() - 1), function.NumFunctions());
 
+  // A random number generator instance to be used for shuffling the order of
+  // visitation.
+  std::mt19937 gen{ std::random_device()() };
```

Comment:
Instead of creating a new generator, we could use the existing: https://github.com/mlpack/mlpack/blob/master/src/mlpack/core/math/random.hpp#L27 That way, we can also use e.g. ``RandomSeed`` to set a static seed for debugging.